keanu-python/keanu/algorithm/sampling.py

```diff
@@ -94,12 +94,14 @@ def sample(net: BayesNet,
            plot: bool = False,
            ax: Any = None) -> sample_types:
 
-    sample_from, sample_from_copy = tee(sample_from)
+    sample_from_copy = list(sample_from)
```

Comment:
Why do they need to be copied?

Reply:
because you can only go through the generator once